internal/rbd/replicationcontrollerserver.go

```diff
@@ -237,20 +259,78 @@ func (rs *ReplicationServer) EnableVolumeReplication(ctx context.Context,
 		}
 	}
 
-	if interval != "" {
-		err = rbdVol.addSnapshotScheduling(interval, startTime)
+	return &replication.EnableVolumeReplicationResponse{}, nil
+}
+
+// getDummyImageName returns the csi-vol-dummy+cluster FSID as the image name.
+// each cluster should have a unique dummy image created. choosing the cluster
+// FSID for the same reason.
+func getDummyImageName(conn *util.ClusterConnection) (string, error) {
+	id, err := conn.GetFSID()
+	if err != nil {
+		return "", err
+	}
+
+	return fmt.Sprintf("csi-vol-dummy-%s", id), nil
+}
+
+// getOperationName returns the operation name for the given operation type
+// combined with the pool name.
+func getOperationName(poolName string, optName operation) string {
+	return fmt.Sprintf("%s/%s", poolName, optName)
+}
+
+// createDummyImage creates a dummy image as a workaround for the rbd
+// scheduling problem.
+func createDummyImage(ctx context.Context, rbdVol *rbdVolume) error {
+	optName := getOperationName(rbdVol.Pool, dummyImageCreated)
+	if _, ok := operationLock.Load(optName); !ok {
+		// create a dummy image
+		imgName, err := getDummyImageName(rbdVol.conn)
 		if err != nil {
-			return nil, err
+			return err
 		}
-		log.DebugLog(
-			ctx,
-			"Added scheduling at interval %s, start time %s for volume %s",
-			interval,
-			startTime,
-			rbdVol)
+		dummyVol := rbdVol
+		dummyVol.RbdImageName = imgName
+		err = createImage(ctx, dummyVol, dummyVol.conn.Creds)
+		if err != nil && !strings.Contains(err.Error(), "File exists") {
+			return err
+		}
+		operationLock.Store(optName, true)
 	}
 
-	return &replication.EnableVolumeReplicationResponse{}, nil
+	return nil
+}
+
+// tickleMirroringOnDummyImage disables and reenables mirroring on the dummy image, and sets a
+// schedule of a minute for the dummy image, to force a schedule refresh for other mirrored images
+// within a minute.
+func tickleMirroringOnDummyImage(rbdVol *rbdVolume, mirroringMode librbd.ImageMirrorMode) error {
+	imgName, err := getDummyImageName(rbdVol.conn)
+	if err != nil {
+		return err
+	}
+	dummyVol := rbdVol
+	dummyVol.RbdImageName = imgName
+
+	dummyImageOpsLock.Lock()
+	defer dummyImageOpsLock.Unlock()
+	err = dummyVol.disableImageMirroring(false)
+	if err != nil {
+		return err
+	}
+
+	err = dummyVol.enableImageMirroring(mirroringMode)
+	if err != nil {
+		return err
+	}
+
+	err = dummyVol.addSnapshotScheduling(admin.Interval("1m"), admin.NoStartTime)
```

Comment:
Technically like create image we can do this call to set the schedule only once and not always. But not optimizing this yet as tests around the DR workflow is completed with this patch.